### PR TITLE
Feature ssl driver options

### PR DESCRIPTION
**Pull request by antoniojlm84**

### CODE REVIEW

**Comment by antoniojlm84** [comment]
why the checks are failing?

File: README.MD [hunk 1]
```diff
@@ -65,6 +65,14 @@ mongo_db_bundle:
         ssl: false
         connectTimeoutMS: 3000 # default null (no timeout)
         readPreference: primaryPreferred # see https://docs.mongodb.com/manual/reference/read-preference/#primary for info
+      
+      ssl_client_name: # see https://docs.mongodb.com/manual/tutorial/configure-ssl-clients/
+        uri: 'mongodb://host1:3062,host2'
+        password: ''
+        authSource: ''
+        ssl: true
+        driver_options:
```
Comment (ilario-pierbattista) [inline README.MD]: Actually I don't agree with this approach.
I'd find a better solution referring to a generic factory for driver options.
The service should implement a specific interface.

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -85,6 +85,12 @@ private function addClients(NodeBuilder $builder)
             ->booleanNode('ssl')
             ->defaultFalse();
 
+        $clientsBuilder
+            ->arrayNode('driver_options')
```
Comment (ilario-pierbattista) [inline src/DependencyInjection/Configuration.php]: Same as above

File: src/DependencyInjection/MongoDbBundleExtension.php [hunk 3]
```diff
@@ -62,6 +62,14 @@ private function defineClientRegistry(array $clientsConfig, bool $debug): void
                 $debug,
             ]
         );
+
+        // Normalize driver options
+        foreach ($clientsConfig as $key => $config) {
+            if (isset($config['driver_options'])) {
+                $clientsConfig[$key]['driver_options'] = $this->normalizeDriverOptions($config);
```
Comment (ilario-pierbattista) [inline src/DependencyInjection/MongoDbBundleExtension.php]: here you would use the factory service

File: src/DependencyInjection/MongoDbBundleExtension.php [hunk 4]
```diff
@@ -93,7 +101,23 @@ private function defineConnections(array $connections): void
...
+    /**
+     * Normalizes the driver options array
+     *
+     * @param array $clientsConfig
+     * @return array|mixed
+     */
+    private function normalizeDriverOptions(array $clientsConfig)
+    {
+        $driverOptions            = $clientsConfig['driver_options'] ?? [];
+        $driverOptions['typeMap'] = ['root' => 'array', 'document' => 'array'];
```
Comment (ilario-pierbattista) [inline src/DependencyInjection/MongoDbBundleExtension.php]: Why?

**Comment by ilario-pierbattista** [comment]
> why the checks are failing?

You should change the tests as well to have them doing the correct assertions.

**Pull request closed without merging**